CONTRIBUTING.adoc

```diff
@@ -373,6 +402,42 @@ The job will:
 3. Compare the two runs.
 4. Post the comparison as a comment to the PR.
 
+=== Merging PRs
+
+==== Merge method and commit history
+
+All 3 Github merge methods (merge commit, squash and merge, and rebase and merge) are allowed.
+Use whichever you deem appropriate.
+As said before, sometimes people use a single PR with multiple commits for their work; other times they create multiple small PRs.
+The best merge method is different for different cases.
+
+That being said, there are not many cases where "rebase and merge" is appropriate.
+And if you use this method, your PR must have a clean commit history: every commit should have a meaningful message, and should be buildable.
+You don't want to have commits like "fix a typo", "this may work" or "wip, done for the day" in master with a linear history.
+And if some of these commits are non-buildable, it can create problems for "git bisect".
```

Comment:
This is basically the reason why I wonder if we should just switch to squash-and-merge by default. In practice lots of "WIP" sneaks in...